Does not compile with non-standard instal of cyrus-sasl

Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ



Description
===========
I have built cyrus-sasl and installed it in a non-standard directory. Building with `-DCMAKE_PREFIX_PATH=<cyrus-sasl install dir>` works for the most part.

But it is missing the `-I` flag for the non-standard directory, which leads to compilation failure:
```
/usr/bin/ccache /usr/lib/ccache/bin/gcc -DLIBRDKAFKA_STATICLIB -I/home/yuval/work/js/sonarw.1/build/rdkafka-prefix/src/rdkafka/src -Igenerated/dummy -isystem /home/yuval/work/js/sonarw.1/build/frontend/openssl-prefix/include -O2 -g -DNDEBUG -fPIC -MD -MT src/CMakeFiles/rdkafka.dir/rdkafka_sasl_cyrus.c.o -MF src/CMakeFiles/rdkafka.dir/rdkafka_sasl_cyrus.c.o.d -o src/CMakeFiles/rdkafka.dir/rdkafka_sasl_cyrus.c.o   -c /home/yuval/work/js/sonarw.1/build/rdkafka-prefix/src/rdkafka/src/rdkafka_sasl_cyrus.c
/home/yuval/work/js/sonarw.1/build/rdkafka-prefix/src/rdkafka/src/rdkafka_sasl_cyrus.c:46:10: fatal error: sasl/sasl.h: No such file or directory
   46 | #include <sasl/sasl.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
```

How to reproduce
================
- remove your sasl package, or simply rename away `/usr/include/sasl/sasl.h`
- build and install cyrus-sasl in a non-standard directory:
  - download and extract https://github.com/cyrusimap/cyrus-sasl/releases/download/cyrus-sasl-2.1.27/cyrus-sasl-2.1.27.tar.gz
  - build using `./configure --prefix /tmp/cyrus-sasl && make && make install`
- try to build rdkafka v1.1.0 with `cmake -DCMAKE_PREFIX_PATH=/tmp/cyrus-sasl`. You will get the above error


**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

The following patch fixes the issue for me:
```
--- rdkafka.orig/src/CMakeLists.txt	2019-08-02 11:40:15.348245463 -0700
+++ rdkafka/src/CMakeLists.txt	2019-08-02 11:50:24.937713227 -0700
@@ -201,6 +201,7 @@
 target_link_libraries(rdkafka PUBLIC Threads::Threads)
 
 if(WITH_SASL_CYRUS)
+  target_include_directories(rdkafka PUBLIC ${SASL_INCLUDE_DIRS})
   target_link_libraries(rdkafka PUBLIC ${SASL_LIBRARIES})
 endif()
```

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `v1.1.0`
 - [x] Apache Kafka version: `N/A`
 - [x] librdkafka client configuration: `N/A`
 - [x] Operating system: `redhat enterprise linux 7, archlinux`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -172,6 +172,7 @@ endif()
 
 if(WITH_ZLIB)
   find_package(ZLIB REQUIRED)
+  target_include_directories(rdkafka PUBLIC ${ZLIB_INCLUDE_DIRS})
   target_link_libraries(rdkafka PUBLIC ZLIB::ZLIB)
 endif()
 
@@ -191,6 +192,7 @@ if(WITH_SSL)
     add_dependencies(rdkafka bundled-ssl)
   else()
     find_package(OpenSSL REQUIRED)
+    target_include_directories(rdkafka PUBLIC ${OPENSSL_INCLUDE_DIR})
     target_link_libraries(rdkafka PUBLIC OpenSSL::SSL OpenSSL::Crypto)
   endif()
 endif()
@@ -203,6 +205,7 @@ find_package(Threads REQUIRED)
 target_link_libraries(rdkafka PUBLIC Threads::Threads)
 
 if(WITH_SASL_CYRUS)
+  target_include_directories(rdkafka PUBLIC ${SASL_INCLUDE_DIRS})
   target_link_libraries(rdkafka PUBLIC ${SASL_LIBRARIES})
 endif()
 
@@ -211,6 +214,7 @@ if(WITH_LIBDL)
 endif()
 
 if(WITH_LZ4_EXT)
+  target_include_directories(rdkafka PUBLIC ${LZ4_INCLUDE_DIRS})
   target_link_libraries(rdkafka PUBLIC LZ4::LZ4)
 endif()
 

